### PR TITLE
feat: experiment with a decorator to hide commands from the help screen

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

This adds a new `@isHidden` [decorator](https://www.typescriptlang.org/docs/handbook/decorators.html) that when applied to a command class will hide it from all `--help` screens. This decorator is being applied to the inactive `oas` and deprecated `swagger` commands.

## 🧬 QA & Testing

* [ ] Running `npm run debug -- --help` you shouldn't see `oas` or `swagger` anywhere.
* [ ] Running `npm run debug -- openapi --help` you shouldn't see `swagger` in the "Related commands" section.

Run `npm run build` and then do the same checks against `./bin/rdme` and you should see the same.

### CODE REVIEW

**Review by kanadgupta — COMMENTED** [review]
this is absolutely wild. wanna add this to the `oas` command as well?

**Comment by erunion** [comment]
Running into all sorts of issues with the `hidden` prop not being picked up in tests. Will keep this draft open until I have time to dig into it.

**Review by kanadgupta — APPROVED** [review]

File: src/lib/commands.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export function getCategories(): Record<
       name: string;
       description: string;
       position: number;
+      hidden: boolean;
```
[the remark this answers is not in the document]
Comment (erunion) [reply src/lib/commands.ts:13]: honestly blown away by how much the rewrite has been helping us this past week

**Pull request merged**